PeriodogramAnalysis/test_power_density_statistics.py

```diff
@@ -0,0 +1,237 @@
+                assert np.isclose(freqs[idx], exp_freq)
+            # Test the psd around peaks comprises more than 90% of total psd
+            assert range_psd_sum / total_psd_sum >= 0.9
+
+
+def test_SpectrumAnalysisPeriodogram() -> None:
+    """
+    Same as previous one.
+    """
+    analysis = SpectrumAnalysisPeriodogram()
+    psd_dict = analysis(mock_signal_generator())
+
+    for channel, channel_data in psd_dict.items():
+        for chunk in range(len(psd_dict[channel]["freqs"])):
+            freqs = channel_data["freqs"][chunk]
+            freqs = np.array(freqs)
+            psd = channel_data["psd"][chunk]
+            total_psd_sum = np.sum(psd)
+            expected_freqs = [5, 10, 15, 20]
+
+            range_psd_sum = 0
+            for exp_freq in expected_freqs:
+                range_psd_sum += np.sum(
+                    psd[(freqs >= exp_freq - 1) & (freqs <= exp_freq + 1)]
+                )
+
+                idx = np.argmin(np.abs(freqs - exp_freq))
+                assert np.isclose(freqs[idx], exp_freq)
+
+            assert range_psd_sum / total_psd_sum >= 0.9
+
+
+def test_SpectrumAnalysisWelch_compute_psd() -> None:
+    """
+    Test compute_psd along, see if compute_psd is working properly.
+    I want to see if this func can append freqs and psd of the new chunk to the end of existing dict.
+
+    I think it is now useless, it duplicates the previous test. I wrote this because coverage that this part is not covered...
+    It should be better to be included in test_SpectrumAnalysisBase_call
+    """
+    analyzer = SpectrumAnalysisWelch()
+    psd_dict_mock: dict[int, dict[str, Any]] = {
+        0: {  # Channel index 0
+            "freqs": [0, 4, 9, 14, 19, 24],
+            "psd": [0, 0.4, 0.5, 0.4, 0.2, 0.1],
+        },
+        1: {  # Channel index 1
+            "freqs": [0, 5, 10, 15, 20, 25],
+            "psd": [0, 0.5, 0.3, 0.2, 0.1, 0],
+        },
+    }
+    signal = next(mock_signal_generator())
+    result = analyzer.compute_psd(signal, psd_dict_mock)
+
+    assert isinstance(result, dict)
+    assert len(result) == 2  # Two channels
+    for channel_index in range(2):  # Two channels
+        assert channel_index in result
+        assert "freqs" in result[channel_index]
+        assert "psd" in result[channel_index]
+        assert len(result[channel_index]["freqs"]) == 7
+        assert len(result[channel_index]["psd"]) == 7
+
+        # Validate numerical values
+        np.testing.assert_array_equal(
+            result[channel_index]["freqs"][:6],
+            psd_dict_mock[channel_index]["freqs"][:6],
+        )
+        np.testing.assert_array_equal(
+            result[channel_index]["psd"][:6], psd_dict_mock[channel_index]["psd"][:6]
+        )
+
+    psd_dict: dict[int, dict[str, Any]] = {}
+    result = analyzer.compute_psd(signal, psd_dict)
+
+    assert isinstance(result, dict)
+    assert len(result) == 2
+
+
+def test_SpectrumAnalysisPeriodogram_compute_psd() -> None:
```

Comment:
Don't copy and paste code

Reply:
Have used pytest.mark.parametrize to get rid of duplication.